fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/RegistrySyncManager.java

```diff
@@ -90,6 +97,38 @@ private static void sendPacket(ServerPlayerEntity player, RegistrySyncPacket pac
 	}
 
 	public static void receivePacket(ThreadExecutor<?> executor, RegistrySyncPacket packet, PacketByteBuf buf, boolean accept, Consumer<Exception> errorHandler) {
+		if (DEBUG) {
+			LOGGER.info("{} raw size: {}", packet.getClass().getSimpleName(), buf.readableBytes());
+
+			// see PacketDeflater
```

Comment:
Hmm, doesn't vanilla deflate custom payloads already? Does deflating again decrease sizes?

Reply:
I don't see a way to know what the current packet is being deflated in the server, so instead I manually compress it in the client instead. I think that's what  JamiesWhiteShirt did on their comparison. The client uncompresses the buffer again.

Reply:
Hmm, if you really want compression, should we build the byte buf and send it into a gzip input stream/output stream for better results, though this might be a silly idea?

Reply:
No, this is done to mimic the vanilla compression so we can measure it. That part is copied directly from vanilla code.